proxy/src/main/java/com/velocitypowered/proxy/connection/client/ClientConfigSessionHandler.java

```diff
@@ -131,6 +132,15 @@ public boolean handle(PingIdentifyPacket packet) {
     return true;
   }
 
+  @Override
+  public boolean handle(KnownPacksPacket packet) {
+    if (player.getConnectionInFlight() != null) {
+      player.getConnectionInFlight().ensureConnected().write(packet);
```

Comment:
This is the important bit of the PR, this ensures that the client's response makes it to the server. (By default Velocity drops unknown packets during the configuration state.)